powerauth-restful-security-base/src/main/java/io/getlime/security/powerauth/rest/api/base/provider/PowerAuthEncryptionProviderBase.java

```diff
@@ -46,36 +59,79 @@
     public abstract PowerAuthEciesDecryptorParameters getEciesDecryptorParameters(String activationId, String applicationKey, String ephemeralPublicKey) throws PowerAuthEncryptionException;
 
     /**
-     * Prepare ECIES data from PowerAuth encryption HTTP header.
+     * Decrypt HTTP request body and construct object with ECIES data. Use the requestType parameter to specify
+     * the type of decrypted object.
      *
-     * @param encryptionHttpHeader PowerAuth encryption HTTP header.
-     * @return PowerAuth ECIES encryption object.
-     * @throws PowerAuthEncryptionException In case PowerAuth encryption HTTP header is invalid.
+     * @param request HTTP request.
+     * @param requestType Request object type.
+     * @return Object with ECIES data.
+     * @throws PowerAuthEncryptionException In case request decryption fails.
      */
-    public PowerAuthEciesEncryption prepareEciesEncryption(String encryptionHttpHeader) throws PowerAuthEncryptionException {
-        // Check for HTTP PowerAuth encryption header
-        if (encryptionHttpHeader == null || encryptionHttpHeader.equals("undefined")) {
-            throw new PowerAuthEncryptionException("POWER_AUTH_ENCRYPTION_INVALID_EMPTY");
+    public <T> PowerAuthEciesEncryption<T> decryptRequest(HttpServletRequest request, Class<T> requestType) throws PowerAuthEncryptionException {
+        // Only POST HTTP method is supported for ECIES
+        if (!"POST".equals(request.getMethod())) {
+            throw new PowerAuthEncryptionException("Invalid HTTP request");
         }
+        // Construct ECIES encryption object from HTTP header
+        final PowerAuthEciesEncryption<T> eciesEncryption = new PowerAuthEciesEncryption<>(request.getHeader(PowerAuthEncryptionHttpHeader.HEADER_NAME));
 
-        // Parse HTTP header
-        PowerAuthEncryptionHttpHeader header = new PowerAuthEncryptionHttpHeader().fromValue(encryptionHttpHeader);
-
-        // Validate the header
         try {
-            PowerAuthEncryptionHttpHeaderValidator.validate(header);
-        } catch (InvalidPowerAuthHttpHeaderException e) {
-            throw new PowerAuthEncryptionException(e.getMessage());
-        }
+            // Parse ECIES cryptogram from request body
+            PowerAuthRequestBody requestBody = ((PowerAuthRequestBody) request.getAttribute(PowerAuthRequestFilterBase.POWERAUTH_REQUEST_BODY));
+            byte[] requestBodyBytes = requestBody.getRequestBytes();
+            if (requestBodyBytes == null || requestBodyBytes.length == 0) {
+                throw new PowerAuthEncryptionException("Invalid HTTP request");
+            }
+            final EciesEncryptedRequest eciesRequest = objectMapper.readValue(requestBodyBytes, EciesEncryptedRequest.class);
```

Comment:
Check for `null`, validate the object...

Reply:
Good catch, I will add the error handling.